examples/RNOneSignal/App.js

```diff
@@ -8,484 +8,700 @@
+     When a notification is received this will fire
+     */
+    onNotificationReceived(notification) {
+        console.log('Notification received: ', notification);
+
+        let debugMsg = 'RECEIVED: \n' + JSON.stringify(notification, null, 2);
+        this.setState({debugText:debugMsg}, () => {
+            console.log("Debug text successfully changed!");
+        });
+    }
+
+    /**
+     When a notification is opened this will fire
+     The openResult will contain information about the notification opened
+     */
+    onNotificationOpened(openResult) {
+        console.log('Message: ', openResult.notification.payload.body);
+        console.log('Data: ', openResult.notification.payload.additionalData);
+        console.log('isActive: ', openResult.notification.isAppInFocus);
+        console.log('openResult: ', openResult);
+
+        let debugMsg = 'OPENED: \n' + JSON.stringify(openResult.notification, null, 2);
+        this.setState({debugText:debugMsg}, () => {
+            console.log("Debug text successfully changed!");
+        });
+    }
+
+    /**
+     Once the user is registered/updated the onIds will send back the userId and pushToken
+        of the device
+     */
+    onIdsAvailable(device) {
+        console.log('Device info: ', device);
+        // Save the userId and pushToken for the device, important for updating the device
+        //  record using the SDK, and sending notifications
+        this.setState({
+            userId: device.userId,
+            pushToken: device.pushToken
+        });
+    }
+
+    /**
+     TODO: Needs to be implemented still in index.js and RNOneSignal.java
+     */
+    onSubscriptionChange(change) {
+        console.log('onSubscriptionChange: ', change);
+    }
+
+    /**
+     TODO: Needs to be implemented still in index.js and RNOneSignal.java
+     */
+    onPermissionChange(change) {
+        console.log('onPermissionChange: ', change);
+    }
+
+    /**
+     Success for the change of state for the email record? or setting subscription state of email record (so logging out)?
+     TODO: Validate functionality and make sure name is correct
+        Should match the onSubscriptionChange and
+
+     TODO: Validate this is working, might be broken after changing name
+     */
+    onEmailSubscriptionChange(change) {
+        console.log('onEmailSubscriptionChange: ', change);
+        this.setState({isEmailLoading:false});
+    }
+
+    /**
+     When an element on an IAM is clicked this will fire
+     The actionResult will contain information about the element clicked
+     */
+    onInAppMessageClicked(actionResult) {
+        console.log('actionResult: ', actionResult);
+
+        let debugMsg = 'CLICKED: \n' + JSON.stringify(actionResult, null, 2);
+        this.setState({debugText:debugMsg}, () => {
+            console.log("Debug text successfully changed!");
+        });
+    }
+
+    render() {
+        return (
+            <ScrollView style={styles.scrollView}>
+
```

Comment:
Remove extra blank lines in this function

Reply:
Spacing will be removed